avocado/core/varianter.py

```diff
@@ -45,9 +45,29 @@ def generate_variant_id(variant):
     :return: String compounded of ordered node names and a hash of all
              values.
     """
+    def variantname(variant):
+        """
+        : param variant: Avocado test variant (list of TreeNode-like objects)
```

Comment:
Missing the summary line + empty line. PEP-257

Reply:
sure, will add